### ISSUE TITLE
Circular Reference if Custom Type is used more than once

### ISSUE THREAD

**Issue by bernard01** [RitwikSaikia/jsonrpc#6]
If a method return type uses a custom data type more than once:

public class Result implements Serializable{
    private SimpleDate date1;
    private SimpleDate date2;  
    ...
}

With a custom data type

public class SimpleDate implements Serializable{
    private int year;
    private int month;
    private int day;
    ...
}

then jsonrpc throws an exception:

java.lang.IllegalArgumentException: circular reference detected : class com.mycompany.SimpleDate
    at org.json.rpc.commons.GsonTypeChecker.isValidType(GsonTypeChecker.java:106)
    at org.json.rpc.commons.GsonTypeChecker.isValidType(GsonTypeChecker.java:128)
    at org.json.rpc.commons.GsonTypeChecker.isValidType(GsonTypeChecker.java:30)
    at org.json.rpc.commons.TypeChecker.isValidMethod(TypeChecker.java:44)
    at org.json.rpc.commons.TypeChecker.isValidInterface(TypeChecker.java:99)
    at org.json.rpc.server.HandleEntry.<init>(HandleEntry.java:53)
    at org.json.rpc.server.JsonRpcExecutor.addHandler(JsonRpcExecutor.java:81)
    at com.mycompany.JsonRpcServlet.bind(JsonRpcServlet.java:25)
    at com.mycompany.JsonRpcServlet.<init>(JsonRpcServlet.java:18)

**Comment by RitwikSaikia** [comment]
will investigate and fix it if an issue, as part of 1.1 release

**Comment by RitwikSaikia** [comment]
You can test out jsonrpc-1.1-SNAPSHOT from the snapshots repo, with the fix.

``` xml
<repositories>
    <repository>
        <id>ritwik-mvn-repo-snapshots</id>
        <url>http://ritwik.net/mvn/snapshots/</url>
    </repository>
</repositories>
```
